src/stcal/ramp_fitting/ramp_fit.py

```diff
@@ -278,3 +279,67 @@ def suppress_one_group_saturated_ramps(ramp_data):
                 ramp_data.groupdq[integ, 0, row, col] = sat_flag
                 sat_pix = (row, col)
                 ramp_data.one_groups[integ].append(sat_pix)
+
+
+def use_zeroframe_for_saturated_ramps(model, dqflags):
+    """
+    For saturated ramps, if there is good data in the ZEROFRAME, replace
+    group zero with the data in ZEROFRAME to use the ramp as a one group
+    ramp.
+
+    Parameters
+    ----------
+    model : data model
+        input data model, assumed to be of type RampModel
+
+    dqflags : dict
+        The data quality flags needed for ramp fitting.
+    """
+    nints, ngroups, nrows, ncols = model.data.shape
+    sat_flag = dqflags["SATURATED"]
+    good_flag = dqflags["GOOD"]
+    dq = model.groupdq
+
+    for integ in range(nints):
+        intdq = dq[integ, :, :, :]
+
+        # Find ramps with a good zeroeth group, but saturated in
+        # the remainder of the ramp.
+        wh_sat = groups_saturated_in_integration(intdq, sat_flag, ngroups)
+
+        whs_rows = wh_sat[0]
+        whs_cols = wh_sat[1]
+        for n in range(len(whs_rows)):
+            row = whs_rows[n]
+            col = whs_cols[n]
+
+            # For ramps completely saturated look for data in the ZEROFRAME
+            # that is non-zero.  If it is non-zero, replace group zero in the
+            # ramp with the data in ZEROFRAME.
+            if model.zeroframe[integ, row, col] != 0:
+                model.data[integ, 0, row, col] = model.zeroframe[integ, row, col]
```

Comment:
Not completely sure that this direct replacement of group zero with zeroframe will eventually give the correct slope value downstream. For cases where only the first group is good, the slope is computed by simply dividing the data value by the group integration time. But because the groups can (and usually will) contain data that've been averaged over several frames (this averaging is done on-board before the data are downlinked), the effective group integration time will be longer than the effective integration time of the zeroframe by itself. Hence the slope may get computed using the wrong integration time.

Reply:
Timing is not handled until the computation of final slopes.  The slope of each integration, not divided by time, is used till the final slope computation.  The final slopes are computed by dividing the integration slope by `effintim`.  This variable is set by `effintim = (nframes + groupgap) * frame_time`.

Reply:
Timing is now properly computed using frame time for `ZEROFRAME` data, rather than group time.